.circleci/config.yml

```diff
@@ -23,81 +21,82 @@ commands:
           name: Grant CircleCI access to dependency locations
           command: |
             sudo chown -R circleci:circleci /usr/local/bin
-            sudo chown -R circleci:circleci /usr/local/lib/<< parameters.python_version >>/site-packages
+            sudo chown -R circleci:circleci /usr/local/lib/python<<parameters.python_version>>/site-packages
       - run:
           name: Install pip setuptools
-          command: python -m pip install -U "pip==21.0.1" "setuptools>=38.0" wheel
+          command: python -m pip install -U "pip>=21.2" "setuptools>=38.0" wheel
       - run:
-          command: |
-            sudo pip install -r test_requirements.txt
+          name: Install test requirements
+          command: sudo pip install -r test_requirements.txt
       - run:
           name: Pip freeze
           command: pip freeze
+
   e2e_run:
     description: Run `kedro run` end to end tests for all starters
     steps:
       - run:
           command: behave features/run.feature
+
   e2e_lint:
     description: Run `kedro lint` end to end tests for all starters
     steps:
       - run:
           command: behave features/lint.feature
 
+
 jobs:
   security_scan:
-    executor: py38
+    parameters:
+      python_version:
+        type: string
+    executor:
+      name: docker
+      python_version: <<parameters.python_version>>
     steps:
       - setup_requirements:
-          python_version: "python3.8"
+          python_version: <<parameters.python_version>>
       - run:
           command: bandit -ll -r .
-  e2e_38:
-    executor: py38
-    steps:
-      - setup_requirements:
-          python_version: "python3.8"
-      - e2e_run
-  e2e_37:
-    executor: py37
-    steps:
-      - setup_requirements:
-          python_version: "python3.7"
-      - e2e_run
-  e2e_36:
-    executor: py36
+
+  e2e_tests:
```

Comment:
Minor: I find the names of these jobs/workflows a little confusing compared to the `command` they execute:
* `lint.feature` ➡️ command `e2e_lint` ➡️ job `lint` 👌 
* `run.feature` ➡️  command `e2e_run` ➡️ job `e2e_tests` -- should be job `run`?

Reply:
I've addressed all comments apart from readme's update but I can easily live with that. Feel free to raise a separate PR for that if you're very keen..